web/src/utils/metrics.ts

```diff
@@ -0,0 +1,191 @@
+    namespace: raw.metric.DstK8S_Namespace,
+    ownerName: raw.metric.DstK8S_OwnerName,
+    ownerType: raw.metric.DstK8S_OwnerType,
+    type: raw.metric.DstK8S_Type,
+    hostName: raw.metric.DstK8S_HostName
+  };
+  source.displayName = buildPeerDisplayName(source, scope);
+  destination.displayName = buildPeerDisplayName(destination, scope);
+  return {
+    source: source,
+    destination: destination,
+    values: raw.values,
+    stats: stats
+  };
+};
+
+/**
+ * computeStats computes avg, max and total. Input metric is always the bytes rate (Bps).
+ */
+export const computeStats = (values: (string | number)[][], range: number | TimeRange): MetricStats => {
+  const filtered = values.map(dp => Number(dp[1])).filter(v => !_.isNaN(v));
+  if (filtered.length === 0) {
+    return { latest: 0, avg: 0, max: 0, total: 0 };
+  }
+
+  // TODO: fill missing dp with 0
+
+  // Figure out what's the expected number of datapoints, because series may not contain all datapoints
+  // We'll assume that missing datapoints are zeros
+  const info = computeStepInterval(range);
+  const rangeInSeconds = rangeToSeconds(range);
+  const expectedDatapoints = Math.floor(rangeInSeconds / info.stepSeconds);
+
+  // Compute stats
+  const sum = filtered.reduce((prev, cur) => prev + cur, 0);
+  const avg = sum / expectedDatapoints;
+  const max = Math.max(...filtered);
+
+  // Get last datapoint. If the serie ends too early before the expected end range, we assume it's 0
+  // (with a tolerance margin)
+  const tolerance = 5 * info.stepSeconds;
+  const endRange = getRangeEnd(range).getTime() / 1000;
+  const lastDP = values[values.length - 1] as [number, string];
+  const latest = lastDP[0] >= endRange - tolerance ? Number(lastDP[1]) : 0;
+
+  return {
+    latest: roundTwoDigits(latest),
+    avg: roundTwoDigits(avg),
+    max: roundTwoDigits(max),
+    total: Math.floor(avg * rangeInSeconds)
+  };
+};
+
+const buildPeerDisplayName = (peer: TopologyMetricPeer, scope: MetricScope): string | undefined => {
+  switch (scope) {
+    case 'host':
+      return peer.hostName ? peer.hostName : peer.type === 'Node' ? peer.name! : undefined;
+    case 'namespace':
+      return peer.namespace ? peer.namespace : peer.type === 'Namespace' ? peer.name! : undefined;
+    case 'owner':
+      return peer.namespace && peer.ownerName
+        ? `${peer.namespace}.${peer.ownerName}`
+        : peer.ownerName
+        ? peer.ownerName
+        : undefined;
+    case 'resource':
+    default:
+      return peer.namespace && peer.name
+        ? `${peer.namespace}.${peer.name}`
+        : peer.name
+        ? peer.name
+        : peer.addr || undefined;
+  }
+};
+
+export const getMetricValue = (v: number, mt: MetricType, mf: MetricFunction): string => {
+  switch (mt) {
+    case 'bytes':
+      return mf === 'sum' ? humanFileSize(v, true, 0) : bytesPerSeconds(v);
+    case 'packets':
+      return mf === 'sum' ? String(v) : elementPerMinText(v);
+  }
+};
```

Comment:
not having default is on the contrary helpful to locate changes that needs to be done, for instance when we add a new type. It's by design: Typescript will raise an error, because it detects not all cases are covered, hence the return type `string` is not satisfied.